src/platform/ESP32/OTAImageProcessorImpl.cpp

```diff
@@ -264,7 +454,23 @@ void OTAImageProcessorImpl::HandleProcessBlock(intptr_t context)
     }
 #endif // CONFIG_ENABLE_ENCRYPTED_OTA
 
-    err = esp_ota_write(imageProcessor->mOTAUpdateHandle, blockToWrite.data(), blockToWrite.size());
+#ifdef CONFIG_ENABLE_DELTA_OTA
+
+    int index = 0;
+    err       = verify_header_data(blockToWrite.data(), blockToWrite.size(), &index);
+
+    if (err != ESP_OK)
+    {
+        ESP_LOGE(TAG, "Header data verification failed (%s)", esp_err_to_name(err));
+        imageProcessor->mDownloader->EndDownload(CHIP_ERROR_INVALID_SIGNATURE);
+        PostOTAStateChangeEvent(DeviceLayer::kOtaDownloadFailed);
+        return;
+    }
+
+    err = esp_delta_ota_feed_patch(imageProcessor->mDeltaOTAUpdateHandle, blockToWrite.data() + index, blockToWrite.size() - index);
```

Comment:
Can you add a comment here... stating that the `esp_delta_ota_feed_patch()` applies the patch and writes that data to the passive partition

Reply:
Done